lib/pbench/server/api/resources/pbench_users.py

```diff
@@ -0,0 +1,449 @@
+                "message": "Already logged in",
+            }
+            return make_response(jsonify(response_object), 200)
+
+        # get the post data
+        post_data = request.get_json()
+        if not post_data:
+            self.logger.warning("Invalid json object: %s", request.url)
+            abort(400, message="Invalid json object in request")
+
+        username = post_data.get("username")
+        if not username:
+            self.logger.warning("Username not provided during the login process")
+            abort(400, message="Please provide a valid username")
+
+        password = post_data.get("password")
+        if not password:
+            self.logger.warning("Password not provided during the login process")
+            abort(400, message="Please provide a valid password")
+
+        try:
+            # fetch the user data
+            user = self.db_session.query(UserModel).filter_by(username=username).first()
+
+        except Exception as e:
+            self.logger.exception("Exception occurred during user login")
+            abort(500, message="INTERNAL ERROR")
+
+        if not user:
+            self.logger.warning(
+                "No user found in the db for Username: {} while login".format(username)
+            )
+            abort(403, message="No such user, please register first")
+
+        # Validate the password
+        if not self.bcrypt.check_password_hash(user.password, password):
+            self.logger.warning(
+                "Wrong password for user: {} while login".format(username)
+            )
+            abort(401, message="Wrong password")
+
+        auth_token = encode_auth_token(self.token_expire_duration, user.id, self.logger)
+        if not auth_token:
+            self.logger.error(
+                "Could not encode the JWT auth token for user: {} while login".format(
+                    username
+                )
+            )
+            # return make_response(jsonify(response_object))
+            abort(
+                500, message="INTERNAL ERROR",
+            )
+
+        """
+        if the user_id is present in the session and a same user tries to log in again we create 
+        another auth token and append it to the existing session id lists of a current user.
+        Otherwise we create a new list of auth_token as a value for the session id key.
+        Our sessions are stored as following key-value pair:
+        session[<user_id>] = [<JWT_Auth_token1>, <JWT_Auth_token2>, ...]
+        Where each token has its own expiry.
+        """
+        if user.id in session:
+            session[user.id].append(auth_token.decode())
+        else:
+            session[user.id] = [auth_token.decode()]
+
+        response_object = {
+            "status": "success",
+            "message": "Successfully logged in.",
+            "auth_token": auth_token.decode(),
+        }
+        return make_response(jsonify(response_object), 200)
```

Comment:
One might argue we're technically creating a session resource, so 201 would be reasonable. I suppose it's not really an issue since we're not (currently) exposing the session resources... however we'll probably want a `get` method so a management console can at least enumerate if not list active sessions, which effectively elevates the internal data to a resource. Ah, well.

Reply:
You mean a `get` method to return all the active sessions?